### ISSUE TITLE
Pipeline annotation causes the same pipeline to be displayed multiple times

### ISSUE THREAD

**Issue by siamaksade** [openshift/origin-web-console#972]
Listing `DeploymentConfig` used in a pipeline with the `pipeline.alpha.openshift.io/uses` annotation makes the same pipeline to be displayed once per each `DeploymentConfig`. That is fine when there is only one `DeploymentConfig` involved but most pipelines involve several objects and that makes the UI very confusing. 

Here is an example when there is a single pipeline that uses multiple `DeploymentConfig` to deploy to Dev, Test, etc:

![pipeline](https://cloud.githubusercontent.com/assets/1174934/20826983/05b0f1de-b86f-11e6-947f-216c94bcbbb5.png)

When the pipeline is triggered, the console displays it 5 times once per each `DeploymentConfig` listed in the `pipeline.alpha.openshift.io/uses` annotation which looks like there are 5 pipelines running simultaneously:

![multiple-pipelines-displayed](https://cloud.githubusercontent.com/assets/1174934/20827005/21abffe6-b86f-11e6-98c9-d61a9031a002.png)

**Comment by siamaksade** [comment]
This is verified on OCP 3.3 and Origin 1.4-rc1.
```
oc v3.3.1.3
kubernetes v1.3.0+52492b4
features: Basic-Auth

Server https://api.ded-int-aws.openshift.com:443
openshift v3.3.1.3
kubernetes v1.3.0+52492b4
```
```
oc v1.4.0-rc1+b4e0954
kubernetes v1.4.0+776c994
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://10.2.2.15:8443
openshift v1.4.0-rc1+b4e0954
kubernetes v1.4.0+776c994
```

**Comment by spadgett** [comment]
Maybe we should display the pipeline once at the top if it triggers multiple deployments. I could see this being common for microservices.

Related to https://trello.com/c/QxbqJsoK

@jwforres @ajacobs21e

**Comment by siamaksade** [comment]
That would definitely make more sense. Showing the same pipeline multiple times would confuse the user.

**Comment by ajacobs21e** [comment]
@siamaksade In your example do you have one app (myapp) that has 5 environments and you're using annotations to group by environment?

**Comment by siamaksade** [comment]
Correct. I have annotated the pipeline (BuildConfig) for myapp (one single app) to list the deploymentconfig of all environments that are deployed to through the pipeline:

```
pipeline.alpha.openshift.io/uses: '[{"name": "myapp-dev", "namespace": "demo", "kind": "DeploymentConfig"},{"name": "myapp-test", "namespace": "demo", "kind": "DeploymentConfig"},{"name": "myapp-uat", "namespace": "demo", "kind": "DeploymentConfig"},{"name": "myapp-stage", "namespace": "demo", "kind": "DeploymentConfig"},{"name": "myapp-preprod", "namespace": "demo", "kind": "DeploymentConfig"}]'
```

In the pipeline, `openshiftDeploy(deploymentConfig: 'dc-name')` is used to trigger a deployment in each env

**Comment by jorgemoralespou** [comment]
This can be a very common use case.

**Comment by spadgett** [comment]
This is addressed by #1335.

There is a List By Pipeline option that shows the pipeline at the top and any deployments below. In other views, the full pipeline is only visible when you expand the section.
